app/packages/api/src/get/quote.ts

```diff
@@ -30,16 +31,41 @@ for await (const record of parser) {
 }
 console.log('Quotes: ', quotes)
 
-export const handler = ApiHandler(async (event: any) => {
+export const handler = ApiHandler(async (event: APIGatewayProxyEventV2) => {
     // @TODO: send quotes already displayed to not send them back again
     // @TODO: add query params to exclude some quotes id
 
-    const id = parseInt(event.pathParameters?.id)
-    console.log('(Optional) ID: ', id)
-    const index = id ? id : Math.floor(Math.random() * quotes.length)
-    console.log('Index: ', index)
+    if (event.pathParameters?.id && !isNaN(parseInt(event.pathParameters.id))) {
+        // if you specify an ID, it won't read the exclude query parameter
+        console.log('ID parameter: ', event.pathParameters.id)
+        const id = parseInt(event.pathParameters.id)
+        if (id < 0 || id >= quotes.length) {
+            return { body: JSON.stringify({ error: 'Invalid ID' }), statusCode: 400 }
+        }
+        return { body: JSON.stringify({ ...quotes[id], id }), statusCode: 200 }
+    }
 
-    const quote = { ...quotes[index], id: index }
+    const exclude: number[] = []
+    if (event.queryStringParameters?.exclude) {
+        console.log('Exclude parameter: ', event.queryStringParameters.exclude)
+        exclude.push(...event.queryStringParameters.exclude.split(',').map(id => parseInt(id)))
+        console.log('Excluding: ', exclude)
+        // check if you there are still quotes to send
+        if (exclude.length >= quotes.length) {
+            console.log('No more quotes to send, ignoring exclude query parameter.')
+            exclude.length = 0
+        }
+    }
+
+    // Get an index that in not in exclude
```

Comment:
C'est pas un vrai random ça ahahah

Reply:
Enfin si, à la première itération

Reply:
Si si c'est un vrai random (enfin je crois), j'ai la preuve si tu veux mais un peu flemme de l'écrire, on s'appelle 😁 TLDR: l'espace des quotes authorisé est plus petit, donc il suffit de récupérer un nombre aléatoire dans cet espace plus petit, ensuite il faut le mapper dans l'espace d'origine (celui avec toutes les quotes), le +1 c'est le mapping (et ça limite la complicité à O(N))